### ISSUE TITLE
Allow multiple headers to be stored in metadata

### ISSUE THREAD

**Issue by JeremyTheocharis** [benthosdev/benthos#1389]
Based on the feedback I got on Discord I am posting this on GitHub as well

## Original request on Discord

Hi, how can I access in benthos an array in a HTTP header response? I want to build a pipeline, which first logs in, recieves a Cookie, and then executes a second command (pls dont ask why it is like that, legacy systems...)

When using 

extract_headers:
        include_prefixes:
          - set-cookie

and then 

headers:
          Cookie: ${! meta("set-cookie") }

I can get only the first cookie. 

I am trying currently to replace a lot of legacy code with a couple of benthos commands, however, this thing is blocking it.

anyone had something similar?
![image](https://user-images.githubusercontent.com/30755722/185205384-0a9a6872-7a34-4a63-b088-e08417eebaa1.png)
![image](https://user-images.githubusercontent.com/30755722/185205406-8ed64377-75e3-4882-90c1-f4d0aaf858bd.png)

(device is offline, so you can do what you want with that PHPSESSID :D)

## Response from Mihai Todor

Hey, I believe the Benthos http components only retrieve the first value from array-based headers: https://github.com/benthosdev/benthos/blob/ba4b1d13570756ac273774d1a2c4772fef18680a/internal/http/client.go#L450 Would you mind opening an issue for this? I'm not sure what's the best way to handle it, but maybe smth like strings.Join("\n", values) instead of values[0] would suit.

**Comment by rasnes** [comment]
We stumbled on this issue ourselves. We have an `x-csrf-token` authentication flow where we need to preserve session state between the request fetching the token and the subsequent requests. That is not possible when I can only get the first value of the response `Set-Cookie` header. "Documentation" for the authentication flow: https://community.sap.com/t5/technology-blogs-by-sap/how-does-csrf-token-work-sap-gateway/bc-p/13520188/highlight/true#M147332

**Comment by rasnes** [comment]
@mihaitodor would you be open for me making a PR for a proposal for a `multi_header` boolean config to the `http` processor? `true` with mean fetching all headers with something like this [here](https://github.com/redpanda-data/benthos/blob/219c3a411197867e97648e5ae26eb864ac1235c5/internal/httpclient/client.go#L215):

```go
func (h *Client) ResponseToBatch(res *http.Response) (service.MessageBatch, error) {
	var resMsg service.MessageBatch

	annotatePart := func(p *service.Message) {
		p.MetaSetMut("http_status_code", res.StatusCode)
		if !h.metaExtractFilter.IsEmpty() {
			for k, values := range res.Header {
				normalisedHeader := strings.ToLower(k)
				if len(values) > 0 && h.metaExtractFilter.Match(normalisedHeader) {
		              var allValues string
		              for i, v := range values {
						  if i > 0 {
							  allValues += "; "
						  }
						  allValues += v
		              }
					 p.MetaSetMut(normalisedHeader, allValues)
				}
			}
		}
	}
	
	...
```

**Comment by rasnes** [comment]
I found a workaround by using `subprocess` calling a `bash` script that obtained all values of the `set-cookie` response header and forwarded it to the subsequent `http` request.

**Comment by mihaitodor** [comment]
@rasnes Yep, that sounds like something worth doing. It would have to be opt-in to avoid breaking existing code. I'd mark the config parameter as advanced, since most people will probably be OK with the current behaviour.
